### PR TITLE
move files to sync subdirectory

### PR DESCRIPTION
**Pull request by jbsv**
First steps to prepare the debugtools package:
- move sources to a subdirectory
- complete the sync API to be aligned to the original

### CODE REVIEW

File: sync/mod.go [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
 //
-// debugsync is using a global logger with some default parameters.
+// sync is using a global logger with some default parameters.
 // It is disabled by default and the level can be increased using
 // an environment variable:
 //
-//	 DBGSYNCLOG=trace
-//	 DBGSYNCLOG=info
+//	 SYNCLOG=trace
+//	 SYNCLOG=info
```
Comment (jbsv) [inline sync/mod.go]: 🐛 in documentation: misaligned with code. need to fix that.

File: sync/mod.go [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
 //
-// debugsync main feature is disabled by default and thus works seemingly
+// sync main feature is disabled by default and thus works seemingly
 // like the original sync package from the standard. To enable the debugging
 // feature, use the following environment variable, e.g:
-//   DBGSYNCON=true
+//   SYNCON=true
```
Comment (jbsv) [inline sync/mod.go]: 🐛 in documentation: misaligned with code. need to fix that.

File: README.md [hunk 3]
```diff
@@ -1,7 +1,13 @@
...
+[![Coverage Status](https://coveralls.io/repos/github/dedis/debugtools/badge.svg?branch=main)](https://coveralls.io/github/dedis/debugsync?branch=main)
+[![Go Report Card](https://goreportcard.com/badge/github.com/dedis/debugtools)](https://goreportcard.com/report/github.com/dedis/debugsync)
+
+# debugtools
+This repository aims to bring useful tools to help debugging applications 
+written in GO.
+
+## sync
+Package that helps debugging mutexes and workgroups in a distributed system. Any
+system deadlock will fire off after a defined timeout.
```
Comment (pierluca) [inline README.md:12]: I'd mention that it's a drop-in replacement for the sync standard library.

File: sync/once.go [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package sync
+
+import (
+	_sync "sync"
+)
+
+// Once is an object that will perform exactly one action.
+// Once is an object that will perform exactly one action.
```
Comment (pierluca) [inline sync/once.go]: twice the comment :-)

**Review by pierluca — CHANGES_REQUESTED** [review]

**Pull request merged**